File: N/1513-number-of-substrings-with-only-1s/main.go
package main

import "fmt"

func numSub(s string) int {
	cnt := 0
	var ret []int
	for _, b := range s {
		if b == '1' {
			cnt++
		} else if cnt > 0 {
			ret = append(ret, cnt)
			cnt = 0
		}
	}
	if cnt > 0 {
		ret = append(ret, cnt)
		cnt = 0
	}
	sum := 0
	for _, c := range ret {
		sum += c * (c + 1) / 2
		sum %= 1000000007
	}
	return sum
}

func main() {
	fmt.Println()
}

File: N/1513-number-of-substrings-with-only-1s/main.rs
impl Solution {
  pub fn num_sub(s: String) -> i32 {
    s.as_bytes()
      .iter()
      .fold((0, 0), |(ans, cnt), &b| {
        if b == b'1' {
          ((ans + cnt + 1) % 1000000007, cnt + 1)
        } else {
          (ans, 0)
        }
      })
      .0
  }
}
